src/cmd/build.rs

```diff
@@ -250,6 +249,222 @@ impl CheckCommand {
     }
 }
 
+const WASM_OPT_INSTALLATION_SUGGESTION: &str =
+                "wasm-opt not found! Make sure the binary is in your PATH environment.\n\n\
+                We use this tool to optimize the size of your contract's Wasm binary.\n\n\
+                wasm-opt is part of the binaryen package. You can find detailed\n\
+                installation instructions on https://github.com/WebAssembly/binaryen#tools.\n\n\
+                There are ready-to-install packages for many platforms:\n\
+                * Debian/Ubuntu: apt-get install binaryen\n\
+                * Homebrew: brew install binaryen\n\
+                * Arch Linux: pacman -S binaryen\n\
+                * Windows: binary releases at https://github.com/WebAssembly/binaryen/releases";
+
+/// A helpful struct for interacting with Binaryen's `wasm-opt` tool.
+struct WasmOptHandler {
+    /// The path to the `wasm-opt` binary.
+    wasm_opt_path: PathBuf,
+    /// The optimization level that should be used when optimizing the Wasm binary.
+    optimization_level: OptimizationPasses,
+    /// Whether or not to keep debugging information in the final Wasm binary.
+    keep_debug_symbols: bool,
+    /// The version number of the `wasm-opt` binary being executed.
+    _version: u32,
```

Comment:
Why prefix with `_`?

Reply:
It's not used in this PR, but will be used in #680